### PR TITLE
Stop showing the view details link for hub template errors

### PR DESCRIPTION
**Pull request by mprahl**
This makes it consistent with the existing user experience.

Relates:
https://issues.redhat.com/browse/ACM-14168

### CODE REVIEW

File: frontend/src/routes/Governance/policies/policy-details/PolicyDetailsResults.tsx [hunk 1]
```diff
@@ -228,15 +241,10 @@ export default function PolicyDetailsResults() {
...
-            const templateExists = !(
-              prunedMessage.includes('Failed to create policy template') ||
-              prunedMessage.includes('check if you have CRD deployed') ||
-              prunedMessage.includes('Dependencies were not satisfied')
-            )
             return (
               <div>
                 {/* message may need to be limited to 300 chars? */}
-                {prunedMessage} {templateExists && templateLink}
+                {prunedMessage} {templateExists(prunedMessage) && templateLink}
```
Comment (yiraeChristineKim) [inline frontend/src/routes/Governance/policies/policy-details/PolicyDetailsResults.tsx:247]: I thought we disabled the link

File: frontend/src/routes/Governance/policies/policy-details/PolicyDetailsResults.tsx [hunk 2]
```diff
@@ -48,6 +48,15 @@ function getTemplateDetailURL(item: ResultsTableData) {
   })
 }
 
+function templateExists(msg: string): boolean {
+  return !(
```
Comment (yiraeChristineKim) [inline frontend/src/routes/Governance/policies/policy-details/PolicyDetailsResults.tsx:52]: Could you review this @JustinKuli @dhaiducek

File: frontend/src/routes/Governance/policies/policy-details/PolicyDetailsResults.tsx (hunk 1 above)
Comment (mprahl) [reply frontend/src/routes/Governance/policies/policy-details/PolicyDetailsResults.tsx:247]: Apparently only if the user doesn't have permission.

File: frontend/src/routes/Governance/policies/policy-details/PolicyDetailsResults.tsx (hunk 2 above)
Comment (JustinKuli) [reply frontend/src/routes/Governance/policies/policy-details/PolicyDetailsResults.tsx:52]: Looks like the "Failed to create policy template" case catches all of the strange situations I could think up. :+1:

**Review by JustinKuli — APPROVED** [review]
/lgtm

**Pull request merged**